File: src/main/java/indi/dkx/laocai/bot/model/segment/MentionSegment.java
package indi.dkx.laocai.bot.model.segment;

import lombok.Data;
import lombok.EqualsAndHashCode;
import indi.dkx.laocai.bot.model.segment.data.MentionSegmentData;

/**
 * mention 提及消息段
 */
@EqualsAndHashCode(callSuper = true)
@Data
public class MentionSegment extends Segment<MentionSegmentData> {
    public MentionSegment (Long userId){
        super("mention", new MentionSegmentData(userId));
    }
}

File: laocai-bot-core/src/main/kotlin/indi/kyson/laocai/bot/core/segment/MentionSegment.kt
package indi.kyson.laocai.bot.core.segment

import com.fasterxml.jackson.annotation.JsonIgnore

/**
 * @提及消息段。
 */
class MentionSegment private constructor(override val data: Data) : Segment {

    data class Data(val userId: Long) {
        fun toSegment(): MentionSegment = MentionSegment(this)
    }

    override val type: String
        get() = "mention"

    @get:JsonIgnore
    val userId: Long
        get() = data.userId

    companion object {
        @JvmStatic
        fun of(userId: Long): MentionSegment = MentionSegment(Data(userId))
    }
}
